pandas/core/frame.py

```diff
@@ -2498,12 +2498,12 @@ def memory_usage(self, index=True, deep=False):
         ...              for t in dtypes])
         >>> df = pd.DataFrame(data)
         >>> df.head()
-           int64  float64  complex128 object  bool
-        0      1      1.0      (1+0j)      1  True
-        1      1      1.0      (1+0j)      1  True
-        2      1      1.0      (1+0j)      1  True
-        3      1      1.0      (1+0j)      1  True
-        4      1      1.0      (1+0j)      1  True
```

Comment:
Is this considered an improvement though?  Look at all of those zero's...

Reply:
There is no `display.precision` set for this, and displaying six digits after the decimal point for a float is default behavior, apparently. I'm not sure why the float64 doesn't appear as `1.000000` rather that `1.0` when generated in that way. When I generate a random float using Numpy `random`, or `random.uniform` I get the default six digit values.

Reply:
@jreback : Thoughts on this?

Reply:
you can step thru the code and see why the float displays as is. I *believe* this is correct. This is a pretty complex interaction of options.

Reply:
did you have a chance to figure this out?

Reply:
A little, yeah. Basically floats' zeros are being removed by the `_trim_zeros` method, where for every digit starting from the right hand side are being removed until we get a decimal or a non-zero number. We would need to modify this function to accept complex numbers as well.

Reply:
I think it might be better to do this for complex as well.

Reply:
Update my PR with the fixes